src/utils.ts

```diff
@@ -102,6 +102,21 @@ export const partitionByKey = <T>(
   return out;
 };
 
+export const partitionByKeySingle = <T>(
+  arr: Array<T>,
+  getKey: (item: T) => Maybe<string>,
+) => {
+  // Important to make partial to denote that for any arbitrary key
+  // there may not be a value.
+  const out: Partial<Record<string, T>> = {};
+  arr.forEach((item) => {
+    const k = getKey(item);
+    if (isNullLike(k)) return;
+    out[k] = item;
```

Comment:
This has last semantics: should we make it first semantics ie out[k] ??= item?

Reply:
sure